decompiler/pipeline/dataflowanalysis/expressionpropagationmemory.py

```diff
@@ -67,3 +78,42 @@ def _update_block_map(self, old_instr_str: str, new_instr_str: str, basic_block:
         self._blocks_map[new_instr_str].add((basic_block, index))
         if (basic_block, index) in self._blocks_map[old_instr_str]:
             self._blocks_map[old_instr_str].remove((basic_block, index))
+
+    def _update_use_map(self, variable: Variable, instruction: Instruction):
+        """
+        Update use map if instruction is changed and a variable is not being used by the instruction anymore:
+        - remove the instruction from the uses-set of the variable
+        - re-add the instruction to the map in order to update uses information for the instruction automatically
+        """
+        if variable not in instruction.requirements:
+            self._use_map.remove_use(variable, instruction)
+            self._use_map.add(instruction)
+
+    def _propagate_postponed_aliased_definitions(self):
+        """
+        Propagate definitions of aliased variables, postponed for propagation, after everything else is propagated.
+        See _is_aliased_postponed_for_propagation method definition for an example why we do not propagate such definitions immediately.
+        """
+        self._initialize_maps(self._cfg)
+        for var in self._postponed_aliased:
+            uses = self._use_map.get(var)
+            definition = self._def_map.get(var)
+            if len(uses) == 1:
+                instruction = uses.pop()
+                if self._is_aliased_postponed_for_propagation(instruction, definition) and self._definition_can_be_propagated(definition):
+                    instruction.substitute(var, definition.value.copy())
+                    self._update_use_map(var, instruction)
+
+    def _definition_can_be_propagated(self, definition: Instruction):
```

Comment:
This function is only used for postponed alias definitions, or? If so, then the definition is always an assignment of one aliased variable to another, so the checks are always fulfilled or do I miss something? So I think we do not need this function.